### PR TITLE
Bump version to 3.24.0 with updated dependencies

### PR DESCRIPTION
**Pull request by kenodegard**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

Supersedes #196 

CC @jezdez 

Added myself as a maintainer

### CODE REVIEW

**Comment by kenodegard** [comment]
@conda-forge-admin, please rerender

**Review by dbast — APPROVED** [review]

**Review by jezdez — APPROVED** [review]

**Comment by dbast** [comment]
My understanding is that `packaging` is added to un-unblock https://github.com/conda-forge/conda-feedstock/pull/202

But *why* is conda-build trying to access `conda/__init__.py` at all while not yet being in the build env (adding packaging to the build env in the conda recipe didn't work)? ... 

what is the call chain here?: e.g. like
* conda-build is called
* conda-build evaluates setup.py not even being in the build-env yet (!?) 
* setup.py imports something out of conda, e.g. due to https://github.com/conda/conda/blame/main/setup.py#L20 
* this lead to `conda/__init__.py` which contains code with (newly) depending on something outside python stdlib (packaging) -> fails 

@kenodegard @jezdez Can you explain this or reference the explanation? (Did not find anything in the conda-build changelog regarding the addition of packaging as conda-build dependency). thanks

**Comment by jaimergp** [comment]
This will probably require a change in conda-smithy that needs to be released, so we can rerender this feedstock. For now, can we check if manually modifying the azure workflows works?

**Comment by jaimergp** [comment]
We might also be able to (mis)use https://conda-forge.org/docs/maintainer/conda_forge_yml.html#remote-ci-setup to add `packaging` there.

**Comment by jaimergp** [comment]
@conda-forge-admin, please rerender

**Review by dbast — APPROVED** [review]

File: conda-forge.yml
Comment (jaimergp) [inline conda-forge.yml:15]: Turns out the key was `remote_ci_setup` (underscores!), so no clue why this worked, or in other words, what else changed that made everything pass. But this change was NOT. We might prefer to revert this to avoid confusion. @conda-forge/conda-build

Comment (jakirkham) [reply conda-forge.yml:15]: Did it work? Not seeing it in CI scripts above. Am I missing something?

**Pull request merged**
